test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -219,6 +230,141 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
 	return nil
 }
 
+// pauseDeployment pauses a Kubernetes deployment using kubectl
+func (data *testData) pauseAndResumeDeployment(namespace, deploymentName string, state string) error {
+	cmd := exec.Command("kubectl", "rollout", state, "deployment", deploymentName, "-n", namespace)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to pause deployment: %v, output: %s", err, string(output))
+	}
+	return nil
+}
+
+// GetOVSPorts returns a list of OVS ports for a given pod.
+func (data *testData) getOVSPorts(targetPod *testPodInfo) ([]string, error) {
+	cmd := []string{"ip", "addr", "show"}
+	stdout, _, err := data.e2eTestData.RunCommandFromPod(data.e2eTestData.GetTestNamespace(), targetPod.podName, containerName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("error listing interfaces for pod %s: %w", targetPod.podName, err)
+	}
+
+	var ovsPorts []string
+	lines := strings.Split(stdout, "\n")
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if len(fields) >= 2 && strings.HasSuffix(fields[0], ":") {
+			ifaceName := strings.Split(strings.TrimSuffix(fields[1], ":"), "@")[0]
+			ovsPorts = append(ovsPorts, ifaceName)
+		}
+	}
+	return ovsPorts, nil
+}
+
+// CheckOVSPortExistence verifies if an OVS port still exists.
+func (data *testData) CheckOVSPortExistence(beforePorts, afterPorts []string) error {
+	beforeSet := make(map[string]struct{})
+	for _, port := range beforePorts {
+		beforeSet[port] = struct{}{}
+	}
+
+	for _, port := range afterPorts {
+		if _, exists := beforeSet[port]; exists {
+			return fmt.Errorf("stale OVS port found after pod deletion: %s", port)
+		}
+	}
+	return nil
+}
+
+// CheckIPReleased verifies if the VLAN IP from the deleted pod is released.
+func (data *testData) checkIPReleased(vlanIP string) error {
+	ipPoolCmd := []string{"kubectl", "get", "ippool", "-o", "jsonpath={.status.allocatedIPs}"}
```

Comment:
I feel it is simpler/better to check the IPPool CR status directly via API. You can check example code in antreaipam_test.go. Sorry, I thought about sending you the pointer earlier but forgot it.

Reply:
Sure , refactoring the code, thanks!